dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -184,23 +184,15 @@ public PagedDoraWorker(
       AlluxioConfiguration conf,
       CacheManager cacheManager,
       MembershipManager membershipManager,
-      BlockMasterClientPool blockMasterClientPool
-  ) {
-    this(workerId, conf, cacheManager, membershipManager, blockMasterClientPool,
-        FileSystemContext.create(conf));
-  }
-
-  protected PagedDoraWorker(
-      AtomicReference<WorkerIdentity> workerId,
-      AlluxioConfiguration conf,
-      CacheManager cacheManager,
-      MembershipManager membershipManager,
       BlockMasterClientPool blockMasterClientPool,
-      FileSystemContext fileSystemContext) {
+      DoraUfsManager ufsManager,
+      DoraMetaManager metaManager,
+      FileSystemContext fileSystemContext
+  ) {
     super(ExecutorServiceFactories.fixedThreadPool("dora-worker-executor", 5));
     mWorkerId = workerId;
     mConf = conf;
-    mUfsManager = mResourceCloser.register(new DoraUfsManager());
+    mUfsManager = mResourceCloser.register(ufsManager);
```

Comment:
I'm just creating the UfsManager outside this constructor and passing it in. The code logic does not change at all so this refactor should be effectively minor.

Reply:
Guice advises against injecting closeable resources especially when it's a singleton, as it's hard to reason about who is responsible to close the resource. https://github.com/google/guice/wiki/Avoid-Injecting-Closable-Resources